src/Common/src/Interop/Unix/libssl/Interop.OpenSsl.cs

```diff
@@ -275,7 +270,7 @@ private static int BioRead(SafeBioHandle bio, IntPtr buffer, int count)
             int bytes = libssl.BIO_read(bio, buffer, count);
             if (bytes != count)
             {
-                throw CreateSslException("Failed in Read BIO");
+                Debug.Fail("BIO read failed.");
```

Comment:
modified to throw exception in BioRead & BioWrite. Removed calling code exception.
Debug.Assert was my first choice as well, but left Debug.Fail because the condition check might come back due to a bio_retry related //todo which I am going to work next. So thought of leaving the same as it is.